src/client.rs

```diff
@@ -142,11 +142,11 @@ impl Client {
         )
         .await
         .expect("Couldn't create client");
-        self.inner = Arc::new(Mutex::new(client));
+        self.inner = Arc::new(client);
```

Comment:
No need to have a mutex here since `Client` has internal mutexes to handle everything "safe" (or at least I expect)

Reply:
Did you test this already?

Reply:
yes, locally and now the workflow is running based on this branch.